Arrowgene.Ddon.GameServer/Chat/Command/Commands/WarpCommand.cs

```diff
@@ -0,0 +1,132 @@
+                    return;
+                }
+            }
+
+            if (command.Length >= 2)
+            {
+                if (Int32.TryParse(command[1], out int parsedLoc))
+                {
+                    startingLocation = parsedLoc;
+                }
+                else
+                {
+                    responses.Add(ChatResponse.CommandError(client, $"Invalid starting location \"{command[1]}\". It must be a number"));
+                    return;
+                }
+            }
+
+            //TODO: Figure out how to send a totally fake quest to the client at the right time so we don't need to hijack a quest file.
+            QuestId baseId = (QuestId)70000001;
+            QuestAssetData baseAsset = _server.AssetRepository.QuestAssets.Quests.Where(x => x.QuestId == baseId).FirstOrDefault();
+            if (baseAsset is null)
+            {
+                responses.Add(ChatResponse.CommandError(client, $"Missing base quest."));
+                return;
+            }
+
+            var questCheckCommands = new List<CDataQuestProcessState.MtTypedArrayCDataQuestCommand>()
+            {
+                new CDataQuestProcessState.MtTypedArrayCDataQuestCommand()
+                {
+                    ResultCommandList = new List<CDataQuestCommand>()
+                    {
+                        new CDataQuestCommand()
+                        {
+                            Command = (ushort)QuestCommandCheckType.StageNo,
+                            Param01 = stageNo
+                        }
+                    },
+                }
+            };
+            var questResultCommands = new List<CDataQuestCommand>()
+            {
+                new CDataQuestCommand()
+                {
+                    Command = (ushort)QuestResultCommand.StageJump,
+                    Param01 = stageNo,
+                    Param02 = startingLocation
+                }
+            };
+
+            //Send fake progress update to trigger the warp command.
+            S2CQuestQuestProgressNtc progressNtc = new S2CQuestQuestProgressNtc()
+            {
+                ProgressCharacterId = client.Character.CharacterId,
+                QuestScheduleId = (uint)baseAsset.QuestScheduleId,
+                QuestProcessStateList = new List<CDataQuestProcessState>()
+                {
+                    new CDataQuestProcessState()
+                    {
+                        ProcessNo = 0,
+                        SequenceNo = 0,
+                        BlockNo = 2,
+                        ResultCommandList = questResultCommands,
+                        CheckCommandList = questCheckCommands
+                    }
+                }
+            };
+            client.Send(progressNtc);
+
+            //Reset quest progress so you can warp again.
+            S2CQuestQuestProgressNtc resetNtc = new S2CQuestQuestProgressNtc()
```

Comment:
I never thought to use this packet like this. This is probably how I send a player back to a quest step when they fail inside a boss room.